web/cpserver.go

```diff
@@ -144,7 +144,7 @@ func (sc *ServiceConfig) Serve(shutdown <-chan (interface{})) {
 		} else {
 			glog.V(2).Infof("httphost: calling uiHandler")
 			if r.TLS == nil {
-				http.Redirect(w, r, fmt.Sprintf("https://%s:%s", r.Host, sc.bindPort), http.StatusMovedPermanently)
+				http.Redirect(w, r, fmt.Sprintf("https://%s:%s", r.Host, strings.Split(sc.bindPort, ":")[1]), http.StatusMovedPermanently)
```

Comment:
We already validate it here: https://github.com/control-center/serviced/blob/develop/validation/validators.go#L92